core/services/gateway/connector/connector.go

```diff
@@ -166,6 +168,9 @@ func (c *gatewayConnector) reconnectLoop(gatewayState *gatewayState) {
 func (c *gatewayConnector) Start(ctx context.Context) error {
 	return c.StartOnce("GatewayConnector", func() error {
 		c.lggr.Info("starting gateway connector")
+		if err := c.handler.Start(ctx); err != nil {
```

Comment:
What is the reason for this `handler.Start` here that wasn't there before?

Reply:
The reason is that I extended GatewayConnectorHandler interface (listed as one of the changes in PR description).